[processor/filter] Update `HasAttrOnDatapoint` and `HasAttrKeyOnDatapoint` examples and docs (#32149)
open-telemetry/opentelemetry-collector-contrib: processor/filterprocessor/README.md

`HasAttrKeyOnDatapoint(key)`

Returns `true` if the given key appears in the attribute map of any datapoint on a metric.

> `key` and `value` must both be strings in the metric and function call. You must use the `metrics.metric` context.

Reviewer: ```suggestion
`key` must be a string. You must use the `metrics.metric` context.
```

Examples:

- `HasAttrKeyOnDatapoint("http.method")`

```yaml
# Drops everything except good metrics
filter/keep_good_metrics:
  error_mode: ignore
  metrics:
    metric:

>       - 'HasAttrKeyOnDatapoint("good.metric") == false'

Reviewer: ```suggestion
      - 'HasAttrKeyOnDatapoint("good.metric")'
```

Converters don't have to be compared to anything if they return a bool

```

#### HasAttrOnDatapoint

`HasAttrOnDatapoint(key, value)`

Returns `true` if the given key and value appears in the attribute map of any datapoint on a metric.

> `key` and `value` must both be strings in the metric and function call. You must use the `metrics.metric` context.

Reviewer: ```suggestion
`key` and `value` must both be strings. If the value of the attribute on the datapoint is not a string, `value` will be compared to `""`. You must use the `metrics.metric` context.
```

Examples:

- `HasAttrOnDatapoint("http.method", "GET")`

```yaml
# Drops everything except good metrics
filter/keep_good_metrics:
  error_mode: ignore
  metrics:
    metric:

>       - 'HasAttrOnDatapoint("good.metric", "true") == false'

Reviewer: ```suggestion
      - 'HasAttrOnDatapoint("good.metric", "true")'
```
